docs/specifications/tests/Basic-TP/basic01.md

```diff
@@ -1,199 +1,339 @@
+      1. Create a [DNS Query] with query type A, *Name Server Name* as query
+         name and RD flag set, and do a [DNS Lookup](#terminology).
+      2. If the [DNS Response], if any, contains a list of A records (follow
+         any CNAME chain) in the answer section then create a set of the IPv4
+         addresses from the A records ("NS IPv4 Addresses").
+      3. Create a [DNS Query] with query type AAAA, *Name Server Name* as
+         query name and RD flag set, and do a [DNS lookup](#terminology).
+      4. If the [DNS Response], if any, contains a list of AAAA records
+         (follow any CNAME chain) in the answer section then create a set of
+         the IPv6 addresses from the AAAA records ("NS IPv6 Addresses").
+      5. If the *NS IPv4 Addresses* set or the *NS IPv6 Addresses* set is
+         non-empty, then then for each IP address in the two sets do if the
+         address is not already listed in the *Parent Name Server IP* set:
+         1. [Send](#terminology) *SOA Child Query* to the IP address.
+         2. If the [DNS Response], if any, meets exactly one of the
+            following two criteria then save the IP address, the parent
+            zone name and the DNS response to the *Parent Name Server IP* set.
+            Criteria:
+            * The [DNS response] is a referral to *Child Zone* (owner name
+              of the NS records in authority section is *Child Zone*), or
+            * The [DNS response] has the AA flag set and the [RCODE Name]
+              NoError.
+
+7. For each name server IP in *Parent Name Server IP* extract
+   parent zone name and the DNS response.
+   1.  If the response is a referral (delegation) to the requested *Child Zone*:
+       1. Save the name server IP and parent zone name to the *Parent Found* set.
+       2. Save the name server IP and parent zone name to the *Delegation Found*
+          set.
+   2.  If the response has the AA flag set and with the [RCODE Name] NXDomain,
+       then
+       1. Save the name server IP and parent zone name to the *Parent Found*
+          set.
+       2. Save the name server IP and parent zone name to the
+          *AA NXDomain Found* set.
+   3.  If the response has the AA flag set and with an SOA record with
+       *Child Zone* as owner name in the answer section, then
+       1. If *Child Zone* is a direct subdomain to parent zone name then save the
+          name server IP and parent zone name to the *Parent Found* set.
+       2. Save the name server IP and parent zone name to the *AA SOA Found* set.
+   4.  If the response has the AA flag set and with a CNAME record with
+       *Child Zone* as owner name in the answer section, then
+       1. Save the name server IP and parent zone name to the *Parent Found* set.
+       2. Save the name server IP and parent zone name to the *AA CNAME Found*
+          set.
+   5.  If response has the AA flag set and with an empty answer section (NODATA),
+       then
```

Comment:
Maybe `If response` -> `If the response`?

Reply:
Fixed (see above).